### PR TITLE
feat: Add graceful shutdown to server with clean connection handling

### PR DESCRIPTION
**Pull request by AnshRaj112**
# Pull Request Description

## Changes Proposed

This PR implements comprehensive graceful shutdown behavior for the PostgreSQL Wire Protocol Mock Server to ensure clean connection termination and proper resource cleanup during server shutdown.

### Key Features Added:

- **Multi-phase shutdown process** with connection draining and client notification
- **Client notification system** that sends shutdown notices to all active connections
- **Automatic transaction rollback** for connections in active transactions
- **Configurable timeouts** for shutdown and connection draining phases
- **Force closure** of lingering connections after timeout expiration
- **Comprehensive status monitoring** with new API methods for shutdown state
- **Signal handling** for SIGTERM, SIGINT, and SIGUSR1
- **Error handling** during shutdown process with graceful degradation
- **Resource cleanup** ensuring all server resources are properly cleaned up

### Technical Implementation:

- Enhanced `ServerManager` class with new shutdown methods and state tracking
- Added configuration options for `shutdownTimeout` and `shutdownDrainTimeout`
- Implemented connection draining with periodic checking and timeout handling
- Added graceful connection closure with proper PostgreSQL protocol messages
- Created comprehensive test suite covering all shutdown scenarios
- Updated documentation with detailed API reference and usage examples

## Related Issues

Issue #58 

## Checklist

- [x] I've read the [contributing guidelines](../CONTRIBUTING.md)
- [x] Tests for the changes have been added/updated
- [x] All tests pass locally
- [x] Documentation has been updated
- [x] The code follows project style guidelines
- [x] No new warnings are generated

## Screenshots (if appropriate)

N/A - This is a backend server enhancement with no UI changes.

### New Configuration Options

The server now supports two new environment variables:
- `PG_MOCK_SHUTDOWN_TIMEOUT` (default: 30000ms) - Total shutdown timeout
- `PG_MOCK_SHUTDOWN_DRAIN_TIMEOUT` (default: 10000ms) - Connection drain timeout

### New API Methods

- `isServerShuttingDown()` - Check if server is currently shutting down
- `getShutdownStatus()` - Get detailed shutdown status information
- `getActiveConnectionCount()` - Get number of active connections
- Enhanced `closeConnection()` with graceful closure support

### Shutdown Process Flow

1. **Shutdown Initiation** - Stop accepting new connections, set shutdown flag
2. **Client Notification** - Send shutdown notices to all active clients
3. **Transaction Rollback** - Automatically rollback any active transactions
4. **Connection Draining** - Wait for clients to disconnect gracefully
5. **Force Closure** - Force close lingering connections after timeout
6. **Resource Cleanup** - Clean up all server resources and state

### Testing

- **22 tests passing** covering all shutdown scenarios
- Tests for basic shutdown, connection handling, draining, error handling
- Configuration and status monitoring tests
- Comprehensive coverage of edge cases and error conditions

### Documentation

- Updated README.md with shutdown configuration and best practices
- Created comprehensive SHUTDOWN.md documentation
- Updated CHANGELOG.md with detailed feature list
- Added API reference and usage examples

This enhancement makes the server production-ready with proper graceful shutdown behavior that ensures data integrity and provides an excellent user experience during server shutdown operations.

### CODE REVIEW

File: SHUTDOWN.md [hunk 1]
```diff
@@ -0,0 +1,363 @@
+# Graceful Shutdown Documentation
```
Comment (bobbyiliev) [inline SHUTDOWN.md:1]: I think we can remove this file! Otherwise it looks good.

**Review by thedevopsdaily — APPROVED** [review]
